### PR TITLE
Add more interceptors tests

### PR DESCRIPTION
**Pull request by RikkiGibson**
Test plan: #67421

Added error when using UnmanagedCallersOnly on an interceptor.
Added warnings when tuple names differ between original and interceptor methods.
Added other misc tests.
Pack errors

The remaining prototype comments are either addressed or moved to the test plan under the relevant items. For example, the prototype comment in MemberSignatureComparer will likely be impacted when we make the generics changes for .NET 8, so I moved it to under a generics bullet in the test plan.

May be a good idea to review d47ca0d8630739c2f72e60f85b555a28268be6c9 standalone as it is just a mechanical packing of the error codes.

### CODE REVIEW

**Comment by RikkiGibson** [comment]
Moving back to draft in order to address prototype comments and get the feature ready for merge to main.

File: src/Compilers/CSharp/Portable/Symbols/Source/SourceMethodSymbolWithAttributes.cs [hunk 1]
```diff
@@ -1234,6 +1242,7 @@ internal override void PostDecodeWellKnownAttributes(ImmutableArray<CSharpAttrib
                         ErrorCode.WRN_ExternMethodNoImplementation;
                     diagnostics.Add(errorCode, this.GetFirstLocation(), this);
                 }
+
```
Comment (RikkiGibson) [inline src/Compilers/CSharp/Portable/Symbols/Source/SourceMethodSymbolWithAttributes.cs:1245]:
```suggestion
```

**Review by cston — APPROVED** [review]

File: src/Compilers/CSharp/Test/Semantic/Semantics/InterceptorsTests.cs [hunk 2]
```diff
@@ -3332,6 +3427,63 @@ class D
...
+    {
+        // Interceptor cannot add a default argument when original method lacks it.
+        var source = """
+            using System.Runtime.CompilerServices;
+            using System;
+
+            C.M(); // 1
+
+            class C
+            {
```
Comment (jcouv) [inline src/Compilers/CSharp/Test/Semantic/Semantics/InterceptorsTests.cs:3468]: nit: unnecessary blank line below

**Review by jcouv — COMMENTED** [review]
Done with review pass (iteration 6)

File: src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs [hunk 3]
```diff
@@ -200,7 +200,7 @@ private void EmbedIfNeedTo(BoundExpression receiver, ImmutableArray<PropertySymb
             {
                 case (not null, null):
                 case (not null, not null) when !methodThisParameter.Type.Equals(interceptorThisParameterForCompare.Type, TypeCompareKind.ObliviousNullableModifierMatchesAny)
-                        || methodThisParameter.RefKind != interceptorThisParameterForCompare.RefKind: // PROTOTYPE(ic): and ref custom modifiers are equal?
+                        || methodThisParameter.RefKind != interceptorThisParameterForCompare.RefKind:
```
[the remark this answers is not in the document]
Comment (RikkiGibson) [reply src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs:203]: No, the prototype comment has been moved to a test plan bullet.

File: src/Compilers/CSharp/Test/Semantic/Semantics/InterceptorsTests.cs [hunk 4]
```diff
@@ -3332,6 +3427,63 @@ class D
...
+
```
Comment (RikkiGibson) [inline src/Compilers/CSharp/Test/Semantic/Semantics/InterceptorsTests.cs:3469]:
```suggestion
```

File: src/Compilers/CSharp/Test/Semantic/Semantics/InterceptorsTests.cs (hunk 2 above)
Comment (RikkiGibson) [reply src/Compilers/CSharp/Test/Semantic/Semantics/InterceptorsTests.cs:3468]: Removed, thanks.

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 7)

**Pull request merged**